<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 4 Colony forming units to determine bacterial counts | ImpactTB/BAA: Standard Operating Procedures for Data Analysis</title>
<meta name="author" content="Colorado State University Coding Team">
<meta name="description" content="4.0.1 Downloads The downloads for this chapter are: Data collection template for recording colony forming units counted on each plate or section of plate in the laboratory Report template to...">
<meta name="generator" content="bookdown 0.39 with bs4_book()">
<meta property="og:title" content="Chapter 4 Colony forming units to determine bacterial counts | ImpactTB/BAA: Standard Operating Procedures for Data Analysis">
<meta property="og:type" content="book">
<meta property="og:description" content="4.0.1 Downloads The downloads for this chapter are: Data collection template for recording colony forming units counted on each plate or section of plate in the laboratory Report template to...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 4 Colony forming units to determine bacterial counts | ImpactTB/BAA: Standard Operating Procedures for Data Analysis">
<meta name="twitter:description" content="4.0.1 Downloads The downloads for this chapter are: Data collection template for recording colony forming units counted on each plate or section of plate in the laboratory Report template to...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.7.0/transition.js"></script><script src="libs/bs3compat-0.7.0/tabs.js"></script><script src="libs/bs3compat-0.7.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">ImpactTB/BAA: Standard Operating Procedures for Data Analysis</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Overview</a></li>
<li><a class="" href="experimental-metadata.html"><span class="header-section-number">2</span> Experimental metadata</a></li>
<li><a class="" href="animal-initial-conditions-and-weekly-weights.html"><span class="header-section-number">3</span> Animal initial conditions and weekly weights</a></li>
<li><a class="active" href="colony-forming-units-to-determine-bacterial-counts.html"><span class="header-section-number">4</span> Colony forming units to determine bacterial counts</a></li>
<li><a class="" href="enzyme-linked-immunosorbest-assay-elisa.html"><span class="header-section-number">5</span> Enzyme-linked immunosorbest assay (ELISA)</a></li>
<li><a class="" href="flow-cytometry.html"><span class="header-section-number">6</span> Flow cytometry</a></li>
<li><a class="" href="pathology.html"><span class="header-section-number">7</span> Pathology</a></li>
<li><a class="" href="proteomics.html"><span class="header-section-number">8</span> Proteomics</a></li>
<li><a class="" href="single-cell-rna-seq.html"><span class="header-section-number">9</span> Single-cell RNA-seq</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="colony-forming-units-to-determine-bacterial-counts" class="section level1" number="4">
<h1>
<span class="header-section-number">4</span> Colony forming units to determine bacterial counts<a class="anchor" aria-label="anchor" href="#colony-forming-units-to-determine-bacterial-counts"><i class="fas fa-link"></i></a>
</h1>
<div id="downloads-1" class="section level3" number="4.0.1">
<h3>
<span class="header-section-number">4.0.1</span> Downloads<a class="anchor" aria-label="anchor" href="#downloads-1"><i class="fas fa-link"></i></a>
</h3>
<p>The downloads for this chapter are:</p>
<ul>
<li>
<a href="https://github.com/csu-impactb/CODING-TEAM-BOOKDOWN-/raw/main/templates/data_collection_templates/cfu_collection_template.xlsx">Data collection template</a> for recording colony forming units counted on each
plate or section of plate in the laboratory</li>
<li>
<a href="https://raw.githubusercontent.com/csu-impactb/CODING-TEAM-BOOKDOWN-/main/templates/report_templates/cfu_report.Rmd">Report template</a> to process data collected with the data template (when you go to this link, go to the “File” bar in your browser’s menu bar, chose “Save As,” then save the file as “animal_weights.Rmd”)</li>
<li>
<a href="https://github.com/csu-impactb/CODING-TEAM-BOOKDOWN-/raw/main/templates/report_templates/cfu_report.docx">Example output</a> from the report template</li>
</ul>
</div>
<div id="overview-test" class="section level3" number="4.0.2">
<h3>
<span class="header-section-number">4.0.2</span> Overview TEST<a class="anchor" aria-label="anchor" href="#overview-test"><i class="fas fa-link"></i></a>
</h3>
<p>In the experiments, we will need to estimate the bacterial load of
<em>Mycobacterium tuberculosis</em> in organs—including lungs and spleens—of
animals from experiments. These measurements help us assess how well a vaccine
has worked in comparison to controls.</p>
<p>We will be estimating bacterial load in an animal organ using the plate count
method with serial dilutions. Serial dilutions allow you to create a
highly diluted sample without needing a massive amount of diluent: as you
increase the dilution one step at a time, you can steadily bring the samples
down to lower bacterial loads per volume. This method is common across
laboratories that study tuberculosis drug efficacy as a method for estimating
bacterial load in animal organs <span class="citation">(<a href="references.html#ref-franzblau2012comprehensive" role="doc-biblioref">Franzblau et al. 2012</a>)</span> and is a
well-established method across microbiology in general, dating back to Koch in
the late 1800s <span class="citation">(<a href="references.html#ref-wilson1922proportion" role="doc-biblioref">Wilson 1922</a>; <a href="references.html#ref-ben2014estimation" role="doc-biblioref">Ben-David and Davidson 2014</a>)</span>.</p>
<p>With this method, we homogenize part of the organ, and then create several
increasingly dilute samples. Each dilution is then spread on a plate with a
medium in which <em>Mycobacterium tuberculosis</em> can grow and left to grow for
several weeks at a temperature conducive to <em>Mycobacterium tuberculosis</em> growth.
The idea is that individual bacteria from the original sample end up randomly
spread across the surface of the plate, and any bacteria that are viable
(able to reproduce) will form a new colony that, after a while, you’ll be able
to see <span class="citation">(<a href="references.html#ref-wilson1922proportion" role="doc-biblioref">Wilson 1922</a>; <a href="references.html#ref-goldman2015practical" role="doc-biblioref">Goldman and Green 2015</a>)</span>. At the end of this
incubation period, you can count the number of these colony-forming units
(CFUs) on each plate.</p>
<p>To count the number of CFUs, you need a “just right” dilution (and
we often won’t know what this is until after plating) to have a countable
plate. If you have too high of a dilution (i.e., one with very few viable
bacteria), randomness will play a big role in the CFU count, and you’ll estimate
the original with more variability, which isn’t ideal. If you have too low
of a dilution (i.e., one with lots of viable bacteria), it will be difficult to
identify separate colonies, and they may compete for resources. (The pattern
you see when the dilution is too low (i.e., too concentrated with bacteria) is
called a lawn—colonies merge together).</p>
<p>Once you identify a good dilution for each sample, the CFU count from this
dilution can be used to estimate the bacterial load in the animal’s organ. To
translate from diluted concentration to original concentration, you do a
back-calculation, incorporating both the number of colonies counted at that
dilution and how dilute the sample was <span class="citation">(<a href="references.html#ref-ben2014estimation" role="doc-biblioref">Ben-David and Davidson 2014</a>; <a href="references.html#ref-goldman2015practical" role="doc-biblioref">Goldman and Green 2015</a>)</span>.</p>
</div>
<div id="template-description-1" class="section level3" number="4.0.3">
<h3>
<span class="header-section-number">4.0.3</span> Template description<a class="anchor" aria-label="anchor" href="#template-description-1"><i class="fas fa-link"></i></a>
</h3>
<p>The data are collected in a spreadsheet with multiple sheets. The first sheet
(named “metadata”) is used to record some metadata for the experiment, while the
following sheets are used to record CFUs counts from the plates used for samples
from each organ, with one sheet per organ. For example, if you plated data
from both the lung and spleen, there would be three sheets in the file: one
with the metadata, one with the plate counts for the lung, and one with the
plate counts for the spleen.</p>
<p>The first sheet, which is the metadata sheet, is shown below:</p>
<div class="inline-figure"><img src="figures/cfu_template_metadata.png" width="100%"></div>
<p>This metadata sheet is used to record information about the overall process of
plating the data. Values from this sheet will be used in calculating the bacterial
load in the original sample based on the CFU counts. This spreadsheet includes
the following columns:</p>
<ul>
<li>
<code>organ</code>: Include one row for each organ that was plated in the experiment.
You should name the organ all in lowercase (e.g., “lung,” “spleen”). You
should use the same name to also name the sheet that records data for that organ
for example, if you have rows in the metadata sheet for “lung” and “spleen,”
then you should have two other sheets in the file, one sheet named “lung” and
one named “spleen,” which you’ll use to store the plate counts for each of those
organs.</li>
<li>
<code>percentage_of_organ</code>: In this column, give the proportion of that organ that
was plated. For example, if you plated half the lung, then in the “lung” row
of this spread sheet, you should put 0.5 in the <code>prop_resuspended</code> column.</li>
<li>
<code>aliquot_ul</code>: 100 uL of the total_resuspended slurry would be considered an original aliquot and is used to peform serial dilutions.</li>
<li>
<code>dilution_factor</code>: Amount of the original stock solution that is present in the
total solution, after dilution(s)</li>
<li>
<code>total_resuspended_mL</code>: This column contains an original volume of tissue homogenate. For example, raw lung tissue is homogenized in 0.5 mL of PBS in a tube containing metal beads.</li>
<li>
<code>volume_plated_ul</code>: Amount of suspension + diluent plated on section of solid agar</li>
</ul>
<p>Following this first sheet in the file, you should have one sheet for each
organ. The organs that you record in these sheets should match up with the rows
on the first, metadata sheet of the template.</p>
<p>Each of these organ-specific sheets should look like this:</p>
<div class="inline-figure"><img src="figures/cfu_organ_sheet_example.png" width="100%"></div>
<p>Each of these organ-specific sheets of the template include the following columns:</p>
<ul>
<li>
<code>count_date</code>: The date that the CFUs were counted. In some cases, the same plates
may be counted at multiple dates.</li>
<li>
<code>who_plated</code>: An identifier for the researcher who plated the sample</li>
<li>
<code>who_counted</code>: An identifier for the researcher who counted the plate on this
specific date</li>
<li>
<code>groups</code>: The experimental group to which the mouse belonged to</li>
<li>
<code>mouse</code>: An identifier for the unique mouse within the group (<em>note: as we
collect data from the new experiment, this can be a unique ID by mouse, based on
notch ID and cage number</em>)</li>
<li>
<code>dil_0</code>, <code>dil_1</code>, <code>dil_2</code>, …: The count at each dilution. You can add additional
columns if there were more dilutions that are in the template or take away
dilution columns if there were fewer. However, all dilution columns should be
named consistently, with “dil_” followed by the dilution number (e.g., “0,” “1,”
“2”). If the CFUs were too numerous to count for a sample at a particular
dilution, put “TNTC” in that cell of the spreadsheet.</li>
</ul>
<p>You can download the template <a href="https://github.com/csu-impactb/CODING-TEAM-BOOKDOWN-/raw/main/templates/data_collection_templates/cfu_collection_template.xlsx">here</a>.
When you download the template, it will have example values filled out in blue.
Use these to get an idea for how to record your own data. When you are ready
to record your own data, delete these example values and replace them with
data collected from your own experiment.</p>
</div>
<div id="processing-collected-data-1" class="section level3" number="4.0.4">
<h3>
<span class="header-section-number">4.0.4</span> Processing collected data<a class="anchor" aria-label="anchor" href="#processing-collected-data-1"><i class="fas fa-link"></i></a>
</h3>
<p>Once data are collected, the file can be run through an R workflow. This workflow
will convert the data into a format that is easier to work with for data analysis
and visualization. It will also produce a report on the data in the spreadsheet, and
ultimately it will also write relevant results in a format that can be used
to populate a global database for all experiments in the project.</p>
<p>The next section provides the details of the pipeline. It aims to explain the
code that processes the data and generates visualizations. You do not need to
run this code step-by-step, but instead can access a script with the full
code <a href="https://raw.githubusercontent.com/csu-impactb/CODING-TEAM-BOOKDOWN-/main/templates/report_templates/cfu_report.Rmd">here</a>.</p>
<p>To use this reporting template, you need to download it to your computer and
save it in the file directory where you saved the data you collected with the
data collection template. You can then open RStudio and navigate so that you are
working within this directory. You should also make sure that you have installed
a few required packages on R on the computer you are using to run the report.
These packages are: <code>tidyverse</code>, <code>readxl</code>, <code>ggbeeswarm</code>, <code>ggpubr</code>, <code>purrr</code>,
<code>knitr</code>, and <code>broom</code>.</p>
<p>Within RStudio, open the report template file. There is one spot where you will
need to change the code in the template file, so it will read in the data from
the version of the template that you saved, which you may have renamed.
In the YAML of the report template file, change the file path beside “data:”
so that it is the file name of your data file.</p>
<p>Once you’ve made this change, you can use the “Knit” button in RStudio to
create a report from the data file and the report template file.</p>
<p>The report includes the following elements:</p>
<ul>
<li>Organ-specific summaries of the experiment, including the number of mice,
experimental groups, date counted, and dilutions used for each experiment</li>
<li>Metadata on the CFU collection process (e.g., percent of organ plated for
each organ, dilution factor)</li>
<li>Plot showing the distribution of CFUs by group in each organ</li>
<li>Table giving the results of an ANOVA analysis comparing log CFUs
across groups within each organ</li>
</ul>
<p>You can download an example of a report created using this template by
clicking <a href="https://github.com/csu-impactb/CODING-TEAM-BOOKDOWN-/raw/main/templates/report_templates/cfu_report.docx">here</a>.</p>
<p>When you knit to create the report, it will create a Word file in the
same file directory where you put your data file and report template.
It will also create and output a version of the data that has been
processed (in the case of the weights data, this mainly involves
tracking mice as they change cages, to link all weights that are from
a single animal). This output fill will be named “cfu_output.csv” and, like
the report file, will be saved in the same file directory as the
data file and the report template.</p>
</div>
<div id="details-of-processing-script-1" class="section level3" number="4.0.5">
<h3>
<span class="header-section-number">4.0.5</span> Details of processing script<a class="anchor" aria-label="anchor" href="#details-of-processing-script-1"><i class="fas fa-link"></i></a>
</h3>
<p>This section goes through the code within the report template. It
explains each part of the code in detail. You do not need to understand
these details to use the report template. However, if you have questions
about how the data are being processed, or how the outputs are created,
all those details are available in this section.</p>
<p>As a note, there are two places in the following code where there’s a small
change compared to the report template. In the report, you incorporate the path
to the data file using the <code>data:</code> section in the YAML at the top of the
document. In the following code, we’ve instead used the path of some example
data within this book’s file directory, so the code will run for this chapter as
well.</p>
<p>First, the workflow loads some additional R libraries. You may need to install
these on your local R session if you do not already have them installed.</p>
<p>Next, the pipeline reads in the organ-specific data. To do this, it creates a
list of all of the sheets that are in the spreadsheet other than the metadata
sheet. It then loops through each of these organ-specific sheets. It uses
pivoting to convert all the dilution levels and values into two columns
(a longer rather than wider format), so that the data from all the organs can
be joined into a single large dataframe, even if a different number of dilutions
were used for the different organs.</p>
<pre><code>## [[1]]
## # A tibble: 880 × 10
##    count_date      who_plated who_counted group mouse day   sex   organ dilution
##    &lt;chr&gt;           &lt;chr&gt;      &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;    &lt;dbl&gt;
##  1 "\"Dec 28 2023… FL         FL          BCG   1     14    fema… lung         0
##  2 "\"Dec 28 2023… FL         FL          BCG   1     14    fema… lung         1
##  3 "\"Dec 28 2023… FL         FL          BCG   1     14    fema… lung         2
##  4 "\"Dec 28 2023… FL         FL          BCG   1     14    fema… lung         3
##  5 "\"Dec 28 2023… FL         FL          BCG   1     14    fema… lung         4
##  6 "\"Dec 28 2023… FL         FL          BCG   1     14    fema… lung         5
##  7 "\"Dec 28 2023… FL         FL          BCG   1     14    fema… lung         6
##  8 "\"Dec 28 2023… FL         FL          BCG   1     14    fema… lung         7
##  9 "\"Dec 28 2023… FL         FL          BCG   1     14    fema… lung         8
## 10 "\"Dec 28 2023… FL         FL          BCG   1     14    fema… lung         9
## # ℹ 870 more rows
## # ℹ 1 more variable: CFUs &lt;chr&gt;
## 
## [[2]]
## # A tibble: 870 × 10
##    count_date      who_plated who_counted group mouse day   sex   organ dilution
##    &lt;chr&gt;           &lt;chr&gt;      &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;    &lt;dbl&gt;
##  1 "\"Dec 28 2023… FL         FL          BCG   1     14    fema… sple…        0
##  2 "\"Dec 28 2023… FL         FL          BCG   1     14    fema… sple…        1
##  3 "\"Dec 28 2023… FL         FL          BCG   1     14    fema… sple…        2
##  4 "\"Dec 28 2023… FL         FL          BCG   1     14    fema… sple…        3
##  5 "\"Dec 28 2023… FL         FL          BCG   1     14    fema… sple…        4
##  6 "\"Dec 28 2023… FL         FL          BCG   1     14    fema… sple…        5
##  7 "\"Dec 28 2023… FL         FL          BCG   1     14    fema… sple…        6
##  8 "\"Dec 28 2023… FL         FL          BCG   1     14    fema… sple…        7
##  9 "\"Dec 28 2023… FL         FL          BCG   1     14    fema… sple…        8
## 10 "\"Dec 28 2023… FL         FL          BCG   1     14    fema… sple…        9
## # ℹ 860 more rows
## # ℹ 1 more variable: CFUs &lt;chr&gt;</code></pre>
<pre><code>## # A tibble: 6 × 10
##   count_date who_plated who_counted group mouse day   sex   organ dilution CFUs 
##   &lt;chr&gt;      &lt;chr&gt;      &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;    &lt;dbl&gt; &lt;chr&gt;
## 1 "\"Dec 28… FL         FL          BCG   1     14    fema… lung         0 TNTC 
## 2 "\"Dec 28… FL         FL          BCG   1     14    fema… lung         1 TNTC 
## 3 "\"Dec 28… FL         FL          BCG   1     14    fema… lung         2 52   
## 4 "\"Dec 28… FL         FL          BCG   1     14    fema… lung         3 10   
## 5 "\"Dec 28… FL         FL          BCG   1     14    fema… lung         4 0    
## 6 "\"Dec 28… FL         FL          BCG   1     14    fema… lung         5 0</code></pre>
<p>At this stage…</p>
<pre><code>## # A tibble: 6 × 4
##   organ  who_plated who_counted count_date       
##   &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt;       &lt;chr&gt;            
## 1 lung   FL         FL          "\"Dec 28 2023\""
## 2 lung   FL         FL          "\"Feb 28 2023\""
## 3 lung   FL         FL          "\"Mar 22 24\""  
## 4 spleen FL         FL          "\"Dec 28 2023\""
## 5 spleen FL         FL          "\"Feb 28 2023\""
## 6 spleen FL         FL          "\"Mar 22 24\""</code></pre>
<pre><code>## # A tibble: 6 × 10
##   count_date who_plated who_counted group mouse day   sex   organ dilution CFUs 
##   &lt;chr&gt;      &lt;chr&gt;      &lt;chr&gt;       &lt;fct&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;    &lt;dbl&gt; &lt;chr&gt;
## 1 "\"Dec 28… FL         FL          BCG   1     14    fema… lung         2 52   
## 2 "\"Dec 28… FL         FL          BCG   1     14    fema… lung         3 10   
## 3 "\"Dec 28… FL         FL          BCG   1     14    fema… lung         4 0    
## 4 "\"Dec 28… FL         FL          BCG   1     14    fema… lung         5 0    
## 5 "\"Dec 28… FL         FL          BCG   1     14    fema… lung         6 0    
## 6 "\"Dec 28… FL         FL          BCG   1     14    fema… lung         7 0</code></pre>
<p>You can see that, rather than having separate columns for each dilution level
on a single row for a sample, there are now multiple rows per sample, with the
CFUs at different dilutions given in a <code>CFUs</code> column, with the <code>dilution</code> column
identifying which dilution level for each.</p>
<p>The next steps work through the data, identifying which dilution is an
appropriate one to use to count CFUs for each sample.</p>
<pre><code>## # A tibble: 171 × 10
##    count_date      who_plated who_counted group mouse day   sex   organ dilution
##    &lt;chr&gt;           &lt;chr&gt;      &lt;chr&gt;       &lt;fct&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;    &lt;dbl&gt;
##  1 "\"Dec 28 2023… FL         FL          Sali… 1     14    fema… lung         2
##  2 "\"Feb 28 2023… FL         FL          Sali… 1     56    fema… lung         2
##  3 "\"Mar 22 24\"" FL         FL          Sali… 1     90    fema… lung         3
##  4 "\"Dec 28 2023… FL         FL          Sali… 1     14    male  lung         3
##  5 "\"Feb 28 2023… FL         FL          Sali… 1     56    male  lung         2
##  6 "\"Mar 22 24\"" FL         FL          Sali… 1     90    male  lung         3
##  7 "\"Dec 28 2023… FL         FL          Sali… 1     14    fema… sple…        0
##  8 "\"Feb 28 2023… FL         FL          Sali… 1     56    fema… sple…        2
##  9 "\"Mar 22 24\"" FL         FL          Sali… 1     90    fema… sple…        2
## 10 "\"Dec 28 2023… FL         FL          Sali… 1     14    male  sple…        1
## # ℹ 161 more rows
## # ℹ 1 more variable: CFUs &lt;dbl&gt;</code></pre>
<p>In the example data, this step has reduced the number observations to consider
from over 1406 to 171.</p>
<pre><code>## [1] 1406</code></pre>
<pre><code>## [1] 171</code></pre>
<p>If you look at the first few rows of the data before and after cleaning, you can
see that in particular it has removed a lot of “TNTC” values (as well as a lot
of 0 values, although that’s harder to see in this sample of the data):</p>
<pre><code>## # A tibble: 5 × 10
##   count_date who_plated who_counted group mouse day   sex   organ dilution CFUs 
##   &lt;chr&gt;      &lt;chr&gt;      &lt;chr&gt;       &lt;fct&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;    &lt;dbl&gt; &lt;chr&gt;
## 1 "\"Dec 28… FL         FL          BCG   1     14    fema… lung         2 52   
## 2 "\"Dec 28… FL         FL          BCG   1     14    fema… lung         3 10   
## 3 "\"Dec 28… FL         FL          BCG   1     14    fema… lung         4 0    
## 4 "\"Dec 28… FL         FL          BCG   1     14    fema… lung         5 0    
## 5 "\"Dec 28… FL         FL          BCG   1     14    fema… lung         6 0</code></pre>
<pre><code>## # A tibble: 5 × 10
##   count_date who_plated who_counted group mouse day   sex   organ dilution  CFUs
##   &lt;chr&gt;      &lt;chr&gt;      &lt;chr&gt;       &lt;fct&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;    &lt;dbl&gt; &lt;dbl&gt;
## 1 "\"Dec 28… FL         FL          Sali… 1     14    fema… lung         2    61
## 2 "\"Feb 28… FL         FL          Sali… 1     56    fema… lung         2    55
## 3 "\"Mar 22… FL         FL          Sali… 1     90    fema… lung         3     4
## 4 "\"Dec 28… FL         FL          Sali… 1     14    male  lung         3    41
## 5 "\"Feb 28… FL         FL          Sali… 1     56    male  lung         2    85</code></pre>
<p>Next, the code brings in the information from the metadata sheet, including data on
what percent of each organ was resuspended, the dilution factor, and so on.
It uses this information to take the CFU value at a given dilution and convert
it to an estimate of CFUs per mL.</p>
<pre><code>## # A tibble: 171 × 11
##    organ count_date      day   who_plated who_counted group sex   mouse dilution
##    &lt;chr&gt; &lt;chr&gt;           &lt;chr&gt; &lt;chr&gt;      &lt;chr&gt;       &lt;fct&gt; &lt;chr&gt; &lt;chr&gt;    &lt;dbl&gt;
##  1 lung  "\"Dec 28 2023… 14    FL         FL          Sali… fema… 1            2
##  2 lung  "\"Feb 28 2023… 56    FL         FL          Sali… fema… 1            2
##  3 lung  "\"Mar 22 24\"" 90    FL         FL          Sali… fema… 1            3
##  4 lung  "\"Dec 28 2023… 14    FL         FL          Sali… male  1            3
##  5 lung  "\"Feb 28 2023… 56    FL         FL          Sali… male  1            2
##  6 lung  "\"Mar 22 24\"" 90    FL         FL          Sali… male  1            3
##  7 lung  "\"Dec 28 2023… 14    FL         FL          Sali… fema… 2            3
##  8 lung  "\"Feb 28 2023… 56    FL         FL          Sali… fema… 2            2
##  9 lung  "\"Mar 22 24\"" 90    FL         FL          Sali… fema… 2            3
## 10 lung  "\"Dec 28 2023… 14    FL         FL          Sali… male  2            3
## # ℹ 161 more rows
## # ℹ 2 more variables: CFUs &lt;dbl&gt;, CFUs_whole &lt;dbl&gt;</code></pre>
<pre><code>## # A tibble: 171 × 11
##    organ count_date      day   who_plated who_counted group sex   mouse dilution
##    &lt;chr&gt; &lt;chr&gt;           &lt;chr&gt; &lt;chr&gt;      &lt;chr&gt;       &lt;fct&gt; &lt;chr&gt; &lt;chr&gt;    &lt;dbl&gt;
##  1 lung  "\"Dec 28 2023… 14    FL         FL          Sali… fema… 1            2
##  2 lung  "\"Feb 28 2023… 56    FL         FL          Sali… fema… 1            2
##  3 lung  "\"Mar 22 24\"" 90    FL         FL          Sali… fema… 1            3
##  4 lung  "\"Dec 28 2023… 14    FL         FL          Sali… male  1            3
##  5 lung  "\"Feb 28 2023… 56    FL         FL          Sali… male  1            2
##  6 lung  "\"Mar 22 24\"" 90    FL         FL          Sali… male  1            3
##  7 lung  "\"Dec 28 2023… 14    FL         FL          Sali… fema… 2            3
##  8 lung  "\"Feb 28 2023… 56    FL         FL          Sali… fema… 2            2
##  9 lung  "\"Mar 22 24\"" 90    FL         FL          Sali… fema… 2            3
## 10 lung  "\"Dec 28 2023… 14    FL         FL          Sali… male  2            3
## # ℹ 161 more rows
## # ℹ 2 more variables: CFUs &lt;dbl&gt;, CFUs_whole &lt;dbl&gt;</code></pre>
<pre><code>## # A tibble: 6 × 11
##   organ count_date day   who_plated who_counted group sex   mouse dilution  CFUs
##   &lt;chr&gt; &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt;      &lt;chr&gt;       &lt;fct&gt; &lt;chr&gt; &lt;chr&gt;    &lt;dbl&gt; &lt;dbl&gt;
## 1 lung  "\"Dec 28… 14    FL         FL          Sali… fema… 1            2    61
## 2 lung  "\"Feb 28… 56    FL         FL          Sali… fema… 1            2    55
## 3 lung  "\"Mar 22… 90    FL         FL          Sali… fema… 1            3     4
## 4 lung  "\"Dec 28… 14    FL         FL          Sali… male  1            3    41
## 5 lung  "\"Feb 28… 56    FL         FL          Sali… male  1            2    85
## 6 lung  "\"Mar 22… 90    FL         FL          Sali… male  1            3     9
## # ℹ 1 more variable: CFUs_whole &lt;dbl&gt;</code></pre>
<p>The rest of the report code is used to provide summaries, visualizations, and
analysis of these data. First, there is code to provide a summary of the number
of mice, experimental groups, and some other details for each of the organs:</p>
<div class="inline-table"><table class="table table-sm">
<caption>
<span id="tab:unnamed-chunk-30">Table 4.1: </span>Organ-specific summary of the experiment</caption>
<thead><tr class="header">
<th align="center">organ</th>
<th align="right">name</th>
<th align="left">value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">lung</td>
<td align="right">Experimental groups:</td>
<td align="left">Saline, BCG, ID93-GLA-SE, BCG+ID93-GLA-SE</td>
</tr>
<tr class="even">
<td align="center">lung</td>
<td align="right">Dates counted:</td>
<td align="left">“Dec 28 2023,” “Feb 28 2023,” “Mar 22 24”</td>
</tr>
<tr class="odd">
<td align="center">lung</td>
<td align="right">Total mice:</td>
<td align="left">17</td>
</tr>
<tr class="even">
<td align="center">lung</td>
<td align="right">Dilutions considered:</td>
<td align="left">2, 3, 4, 5</td>
</tr>
<tr class="odd">
<td align="center">spleen</td>
<td align="right">Experimental groups:</td>
<td align="left">Saline, BCG, ID93-GLA-SE, BCG+ID93-GLA-SE</td>
</tr>
<tr class="even">
<td align="center">spleen</td>
<td align="right">Dates counted:</td>
<td align="left">“Dec 28 2023,” “Feb 28 2023,” “Mar 22 24”</td>
</tr>
<tr class="odd">
<td align="center">spleen</td>
<td align="right">Total mice:</td>
<td align="left">17</td>
</tr>
<tr class="even">
<td align="center">spleen</td>
<td align="right">Dilutions considered:</td>
<td align="left">0, 1, 2, 3</td>
</tr>
</tbody>
</table></div>
<p>Next, the pipeline provides a table with the conditions of the CFU collection,
based on the collected metadata from the template:</p>
<div class="inline-table"><table class="table table-sm">
<caption>
<span id="tab:unnamed-chunk-31">Table 4.2: </span>Conditions of the CFU collection</caption>
<colgroup>
<col width="8%">
<col width="27%">
<col width="18%">
<col width="25%">
<col width="19%">
</colgroup>
<thead><tr class="header">
<th align="left">Organ</th>
<th align="right">Percent of Organ Plated</th>
<th align="right">Dilution Factor</th>
<th align="right">Total resuspension mL</th>
<th align="right">Volume Plated mL</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">lung</td>
<td align="right">33</td>
<td align="right">10</td>
<td align="right">1.5</td>
<td align="right">0.1</td>
</tr>
<tr class="even">
<td align="left">spleen</td>
<td align="right">50</td>
<td align="right">10</td>
<td align="right">1.5</td>
<td align="right">0.1</td>
</tr>
</tbody>
</table></div>
<p>Next, the pipeline creates a plot showing the distribution of CFUs by
experimental group in each of the organs:</p>
<div class="inline-figure"><img src="csu-impactb_files/figure-html/unnamed-chunk-32-1.png" width="480"></div>
<div class="inline-figure"><img src="csu-impactb_files/figure-html/unnamed-chunk-33-1.png" width="672"></div>
<p>Next, the pipeline runs an ANOVA analysis on the data. This is conducted after
transforming the CFUs with a log-10 transform.</p>
<div class="inline-table"><table class="table table-sm">
<caption>
<span id="tab:unnamed-chunk-34">Table 4.3: </span>ANOVA results comparing CFUs in each organ across the experimental groups</caption>
<thead><tr class="header">
<th align="left">organ</th>
<th align="left">contrast1</th>
<th align="left">contrast2</th>
<th align="right">estimate</th>
<th align="right">conf.low</th>
<th align="right">conf.high</th>
<th align="right">adj.p.value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">lung</td>
<td align="left">BCG</td>
<td align="left">Saline</td>
<td align="right">-0.2368058</td>
<td align="right">-0.5832132</td>
<td align="right">0.1096016</td>
<td align="right">0.2844430</td>
</tr>
<tr class="even">
<td align="left">lung</td>
<td align="left">ID93</td>
<td align="left">GLA</td>
<td align="right">0.0407358</td>
<td align="right">-0.2933692</td>
<td align="right">0.3748408</td>
<td align="right">0.9886291</td>
</tr>
<tr class="odd">
<td align="left">lung</td>
<td align="left">BCG+ID93</td>
<td align="left">GLA</td>
<td align="right">-0.2381384</td>
<td align="right">-0.5760187</td>
<td align="right">0.0997419</td>
<td align="right">0.2587702</td>
</tr>
<tr class="even">
<td align="left">lung</td>
<td align="left">ID93</td>
<td align="left">GLA</td>
<td align="right">0.2775416</td>
<td align="right">-0.0688658</td>
<td align="right">0.6239490</td>
<td align="right">0.1614924</td>
</tr>
<tr class="odd">
<td align="left">lung</td>
<td align="left">BCG+ID93</td>
<td align="left">GLA</td>
<td align="right">-0.0013326</td>
<td align="right">-0.3513826</td>
<td align="right">0.3487175</td>
<td align="right">0.9999996</td>
</tr>
<tr class="even">
<td align="left">lung</td>
<td align="left">BCG+ID93</td>
<td align="left">GLA</td>
<td align="right">-0.2788742</td>
<td align="right">-0.6167545</td>
<td align="right">0.0590061</td>
<td align="right">0.1420190</td>
</tr>
<tr class="odd">
<td align="left">spleen</td>
<td align="left">BCG</td>
<td align="left">Saline</td>
<td align="right">-0.2738479</td>
<td align="right">-1.0919602</td>
<td align="right">0.5442645</td>
<td align="right">0.8159594</td>
</tr>
<tr class="even">
<td align="left">spleen</td>
<td align="left">ID93</td>
<td align="left">GLA</td>
<td align="right">-0.0292823</td>
<td align="right">-0.8276782</td>
<td align="right">0.7691136</td>
<td align="right">0.9996778</td>
</tr>
<tr class="odd">
<td align="left">spleen</td>
<td align="left">BCG+ID93</td>
<td align="left">GLA</td>
<td align="right">-0.4036624</td>
<td align="right">-1.2329754</td>
<td align="right">0.4256505</td>
<td align="right">0.5798709</td>
</tr>
<tr class="even">
<td align="left">spleen</td>
<td align="left">ID93</td>
<td align="left">GLA</td>
<td align="right">0.2445656</td>
<td align="right">-0.5735467</td>
<td align="right">1.0626780</td>
<td align="right">0.8612381</td>
</tr>
<tr class="odd">
<td align="left">spleen</td>
<td align="left">BCG+ID93</td>
<td align="left">GLA</td>
<td align="right">-0.1298145</td>
<td align="right">-0.9781256</td>
<td align="right">0.7184965</td>
<td align="right">0.9779295</td>
</tr>
<tr class="even">
<td align="left">spleen</td>
<td align="left">BCG+ID93</td>
<td align="left">GLA</td>
<td align="right">-0.3743802</td>
<td align="right">-1.2036931</td>
<td align="right">0.4549328</td>
<td align="right">0.6381738</td>
</tr>
</tbody>
</table></div>
<pre><code>## [1] 0.6506753</code></pre>
<pre><code>## [1] 0.8720107</code></pre>
<pre><code>## [1] 0.3555729</code></pre>
<pre><code>## [1] 0.01680341</code></pre>
<pre><code>## [1] 0.2140449</code></pre>
<pre><code>## [1] 0.0267285</code></pre>
<p>As a last step, the code in the template writes a CSV file with the processed
data. This file will be an input into a script that will format the data to
add to a database where we are collecting and integrating data from all the CSU
experiments, and ultimately from there into project-wide storage.</p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="animal-initial-conditions-and-weekly-weights.html"><span class="header-section-number">3</span> Animal initial conditions and weekly weights</a></div>
<div class="next"><a href="enzyme-linked-immunosorbest-assay-elisa.html"><span class="header-section-number">5</span> Enzyme-linked immunosorbest assay (ELISA)</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav"><li>
<a class="nav-link" href="#colony-forming-units-to-determine-bacterial-counts"><span class="header-section-number">4</span> Colony forming units to determine bacterial counts</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#downloads-1"><span class="header-section-number">4.0.1</span> Downloads</a></li>
<li><a class="nav-link" href="#overview-test"><span class="header-section-number">4.0.2</span> Overview TEST</a></li>
<li><a class="nav-link" href="#template-description-1"><span class="header-section-number">4.0.3</span> Template description</a></li>
<li><a class="nav-link" href="#processing-collected-data-1"><span class="header-section-number">4.0.4</span> Processing collected data</a></li>
<li><a class="nav-link" href="#details-of-processing-script-1"><span class="header-section-number">4.0.5</span> Details of processing script</a></li>
</ul>
</li></ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>ImpactTB/BAA: Standard Operating Procedures for Data Analysis</strong>" was written by Colorado State University Coding Team. It was last built on 2024-06-21.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
