<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 5 Enzyme-linked immunosorbest assay (ELISA) | ImpactTB/BAA: Standard Operating Procedures for Data Analysis</title>
<meta name="author" content="Colorado State University Coding Team">
<meta name="description" content="ELISA is a standard molecular biology assay for detecting and quantifying a variety of compounds, including peptides, proteins, and antibodies in a sample. The sample could be serum, plasma, or...">
<meta name="generator" content="bookdown 0.39 with bs4_book()">
<meta property="og:title" content="Chapter 5 Enzyme-linked immunosorbest assay (ELISA) | ImpactTB/BAA: Standard Operating Procedures for Data Analysis">
<meta property="og:type" content="book">
<meta property="og:description" content="ELISA is a standard molecular biology assay for detecting and quantifying a variety of compounds, including peptides, proteins, and antibodies in a sample. The sample could be serum, plasma, or...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 5 Enzyme-linked immunosorbest assay (ELISA) | ImpactTB/BAA: Standard Operating Procedures for Data Analysis">
<meta name="twitter:description" content="ELISA is a standard molecular biology assay for detecting and quantifying a variety of compounds, including peptides, proteins, and antibodies in a sample. The sample could be serum, plasma, or...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.7.0/transition.js"></script><script src="libs/bs3compat-0.7.0/tabs.js"></script><script src="libs/bs3compat-0.7.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">ImpactTB/BAA: Standard Operating Procedures for Data Analysis</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Overview</a></li>
<li><a class="" href="experimental-metadata.html"><span class="header-section-number">2</span> Experimental metadata</a></li>
<li><a class="" href="animal-initial-conditions-and-weekly-weights.html"><span class="header-section-number">3</span> Animal initial conditions and weekly weights</a></li>
<li><a class="" href="colony-forming-units-to-determine-bacterial-counts.html"><span class="header-section-number">4</span> Colony forming units to determine bacterial counts</a></li>
<li><a class="active" href="enzyme-linked-immunosorbest-assay-elisa.html"><span class="header-section-number">5</span> Enzyme-linked immunosorbest assay (ELISA)</a></li>
<li><a class="" href="flow-cytometry.html"><span class="header-section-number">6</span> Flow cytometry</a></li>
<li><a class="" href="pathology.html"><span class="header-section-number">7</span> Pathology</a></li>
<li><a class="" href="proteomics.html"><span class="header-section-number">8</span> Proteomics</a></li>
<li><a class="" href="single-cell-rna-seq.html"><span class="header-section-number">9</span> Single-cell RNA-seq</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="enzyme-linked-immunosorbest-assay-elisa" class="section level1" number="5">
<h1>
<span class="header-section-number">5</span> Enzyme-linked immunosorbest assay (ELISA)<a class="anchor" aria-label="anchor" href="#enzyme-linked-immunosorbest-assay-elisa"><i class="fas fa-link"></i></a>
</h1>
<p>ELISA is a standard molecular biology assay for detecting and quantifying a
variety of compounds, including peptides, proteins, and antibodies in a sample.
The sample could be serum, plasma, or bronchoalveolar lavage fluid (BALF).</p>
<div id="importance-of-elisa" class="section level2" number="5.1">
<h2>
<span class="header-section-number">5.1</span> <strong>Importance of ELISA</strong><a class="anchor" aria-label="anchor" href="#importance-of-elisa"><i class="fas fa-link"></i></a>
</h2>
<p>An antigen-specific reaction in the host results in the production of antibodies, which are proteins found in the blood. In the event of an infectious disease,
it aids in the detection of antibodies in the body. ELISA is distinguishable from other antibody-assays in that it produces quantifiable findings and separates non-specific from specific interactions by serial binding to solid surfaces,
which is often a polystyrene multi-well plate.</p>
<p>In IMPAc-TB project, it is crucial to evaluate the if the vaccine is eliciting humoral immunity and generating antibodies against vaccine antigen. ELISA will
be used to determine the presence of Immunoglobulin (Ig) IgG, IgA, and IgM in
the serum different time points post-vaccination.</p>
<div id="principle-of-elisa" class="section level3" number="5.1.1">
<h3>
<span class="header-section-number">5.1.1</span> <strong>Principle of ELISA</strong><a class="anchor" aria-label="anchor" href="#principle-of-elisa"><i class="fas fa-link"></i></a>
</h3>
<p>ELISA is based on the principle of antigen-antibody interaction. An antigen must be immobilized on a solid surface and then complexed with an enzyme-linked antibody in an ELISA. The conjugated enzyme’s activity is evaluated by incubating it with a substrate to yield a quantifiable result, which enables detection. There are four basic steps of ELISA:</p>
<p><strong>1. Coating multiwell plate with antigen/antibody</strong>: This step depends on what we want to detect the sample. If we need to evaluate the the presence of antibody, the plate will be coated with the antigen, and vice versa. To coat the plate, a fixed concentration of antigen (protein) is added to a 96 well high-binding plate (charged plate). Plate is incubated over night with the antigen at 4 degree celsius (as proteins are temperature sensitive) so that antigens are completely bound to the well.</p>
<p><strong>2. Blocking</strong>: It is possible that not each and every site of the well is coated with the targeted antigen, and there could be uncovered areas. It is important to block those empty spaces so that primary antibody (which we will add to the next step) binds to these spaces and give us false positive results. For this, microplate well surface-binding sites are blocked with an unrelated protein or other substance.Most common blocking agents are bovine serum albumin, skim milk, and casein. One of the best blocking agents is to use the serum from the organism in which your secondary (detection antibody) is raised. For example, if the secondary antibody is raised in goat, then we can use goat serum as a blocking agent.</p>
<p><strong>3. Probing</strong>: Probing is the step where we add sample containing antibodies that we want to detect. This will be the primary antibody. If the antibodies against the antigen (which we have coated) are present in the sample, it will bind to the antigen with high affinity.</p>
<p><strong>4. Washing</strong>: After the incubation of sample containing primary antibody, the wells are washed so that any unbound antibody is washed away. Washing solution contains phosphate buffer saline + 0.05% tween-20 (a mild detergent). 0.05% tween-20 washes away all the non-specific interactions as those are not strong, but keeps all the specific interaction as those are strong and cannot be detached with mild detergent.</p>
<p><strong>5. Detection</strong>: To detect the presence of antibody-antigen complex, a secondary antibody labelled with an enzyme (usually horseradish peroxidase) is added to the wells, incubated and washed.</p>
<p><strong>6. Signal Measurement</strong>: Finally to detect “if” and “how much” of the antibody is present, a chromogenic substrate (like 3,3’,5,5’-Tetramethylbenzidine) is added to the wells, which can be cleaved the the enzyme that is tagged to the secondary antibody. The color compund is formed after the addition of the substrate, which is directly proportional to the amount of antibody present in the sample. The plate is read on a plate reader, where color is converted to numbers.</p>
<div class="figure">
<span style="display:block;" id="fig:pressure"></span>
<img src="DATA/elisa.png" alt="A caption" width="100%"><p class="caption">
Figure 5.1: A caption
</p>
</div>
</div>
<div id="loading-libraries" class="section level3" number="5.1.2">
<h3>
<span class="header-section-number">5.1.2</span> Loading libraries<a class="anchor" aria-label="anchor" href="#loading-libraries"><i class="fas fa-link"></i></a>
</h3>
</div>
</div>
<div id="elisa-data-analysis" class="section level2" number="5.2">
<h2>
<span class="header-section-number">5.2</span> ELISA data analysis<a class="anchor" aria-label="anchor" href="#elisa-data-analysis"><i class="fas fa-link"></i></a>
</h2>
<p>Analysis of ELISA data is the most important part of the ELISA experiment. ELISA data can be analyzed in different ways based on how the data is acquired. There are a a few examples of the type of ELISA data :</p>
<p><strong>1. With standard curve:</strong> ELISA can be used to determine the concentrations of the antigen and antibody. This type of ELISA data usually have a standard curve with
different concentrations of the known analyte and the concentration in the sample is determined by extrapolating the unknown values in the curve. This type of assay is straightforward, easy to interpret and are more robust.</p>
<p><strong>2. Without standard curve:</strong> Usually vaccine studies involve investigating the presence of high-affinity (and novel) antibodies against the vaccine antigens.
Therefore, plotting a standard curve is not feasible as there is no previous information available for antibody concentration or type of antibody. Also, because antibody response to a vaccine will differ depending on the individual,
it is not practical to generate a calibration curve from which absolute concentrations can be extrapolated.
For this type of ELISA, quantification of the antibody titers is performed using serial dilutions of the test samples, and analysis can be performed using the following three methods <span class="citation">(<a href="references.html#ref-hartman2018absorbance" role="doc-biblioref">Hartman et al. 2018</a>)</span>:</p>
<ol style="list-style-type: decimal">
<li>Fitting sigmoid model</li>
<li>Endpoint titer method
3: Absorbance summation method</li>
</ol>
<p>Let’s have a look at these methods, how we can apply these methods in our data, and R-based packages that we can utilize to perform this analysis.</p>
</div>
<div id="curve-fitting-model" class="section level2" number="5.3">
<h2>
<span class="header-section-number">5.3</span> <strong>1. Curve fitting model:</strong><a class="anchor" aria-label="anchor" href="#curve-fitting-model"><i class="fas fa-link"></i></a>
</h2>
<p>The curve in ELISA data represents a plot of known concentrations versus their corresponding signal responses. The typical range of these calibration curves is one to two orders of magnitude on the response axis and two or more orders of magnitude on the concentration axis. The real curve of each assay could be easily identified if an infinite number of concentration dilutions with an infinite number of repetitions could be tested. The correct curve must be approximated from a relatively small number of noisy points, though, because there are a finite number of dilutions that may be performed. To estimate the dose-response relationship between standard dilutions, a method of interpolating between standards is required because there cannot be a standard at every concentration. This process is typically performed using a mathematical function or regression to approximate the true shape of the curve. A curve model is the name given to this approximating function, which commonly uses two or more parameters to describe a family of curves, and are then adjusted in order to find the curve from the family of curves that best fits the assay data.</p>
<p>Three qualities should be included in a good curve fitting model.
1. The true curve’s shape must be accurately approximated by the curve model. If the curve model does not accomplish this, there is no way to adjust for this component of the total error that results from a lack of fit.
2. In order to get concentration estimates with minimal inaccuracy, a decent curve model must be able to average away as much of the random variation as is practical. 3. A successful curve model must be capable of accurately predicting concentration values for points between the anchor points of the standard dilutions.</p>
<div id="how-do-we-perform-curve-fitting-model" class="section level3" number="5.3.1">
<h3>
<span class="header-section-number">5.3.1</span> How do we perform curve fitting model<a class="anchor" aria-label="anchor" href="#how-do-we-perform-curve-fitting-model"><i class="fas fa-link"></i></a>
</h3>
<p>There are two major steps in performing curve fitting model for non-linear data like ELISA:
1. Finding the initial starting estimates of the parameters
2. locating the optimal solution in a region of the initial estimates</p>
<p>We have presented an example below where we have performed a 8-10 point serial dilution of our sample and fitted a 4 parameter curve model.</p>
</div>
<div id="an-example-of-the-curve-fitting-model" class="section level3" number="5.3.2">
<h3>
<span class="header-section-number">5.3.2</span> An example of the curve fitting model<a class="anchor" aria-label="anchor" href="#an-example-of-the-curve-fitting-model"><i class="fas fa-link"></i></a>
</h3>
<div id="read-in-the-data" class="section level4" number="5.3.2.1">
<h4>
<span class="header-section-number">5.3.2.1</span> Read in the data<a class="anchor" aria-label="anchor" href="#read-in-the-data"><i class="fas fa-link"></i></a>
</h4>
<p>This information comes from the 2018 study conducted by Hartman et al. Hartman et al. analyzed the ELISA data in their study utilizing fitted sigmoid analysis, end point titer, and absorbance summation. We utilized this information to determine whether our formulas and calculations provide the same outcomes and values as theirs.</p>
</div>
<div id="tidying-the-data" class="section level4" number="5.3.2.2">
<h4>
<span class="header-section-number">5.3.2.2</span> Tidying the data<a class="anchor" aria-label="anchor" href="#tidying-the-data"><i class="fas fa-link"></i></a>
</h4>
<p>We next performed tidying the data and make it in a format so that we can plot a sigmoid curve with that.</p>
<pre><code>## # A tibble: 6 × 5
##   numerator denominator absorbance dilution log_dilution
##       &lt;dbl&gt;       &lt;dbl&gt;      &lt;dbl&gt;    &lt;dbl&gt;        &lt;dbl&gt;
## 1         1          30       4    0.0333          -3.10
## 2         1          90       3.73 0.0111          -4.10
## 3         1         270       2.34 0.00370         -5.10
## 4         1         810       1.1  0.00123         -6.10
## 5         1        2430       0.51 0.000412        -7.10
## 6         1        7290       0.22 0.000137        -8.10</code></pre>
</div>
<div id="create-function-for-curve-fitting-model" class="section level4" number="5.3.2.3">
<h4>
<span class="header-section-number">5.3.2.3</span> Create function for curve fitting model<a class="anchor" aria-label="anchor" href="#create-function-for-curve-fitting-model"><i class="fas fa-link"></i></a>
</h4>
<p>We next created the curve fitting model function by using nlsLM function from “minpack.lm” package.
The purpose of nlslm is to minimize the sum square of the vector returned by the function fn, by a modification of the Levenberg-Marquardt algorithm. In the early 1960s, the Levenberg-Marquardt algorithm was developed to address nonlinear least squares problems. Through a series of well-chosen updates to model parameter values, Levenberg-Marquardt algorithm lower the sum of the squares of the errors between the model function and the data points.</p>
<pre><code>## Nonlinear regression model
##   model: absorbance ~ ((a - d)/(1 + (log_dilution/c)^b)) + d
##    data: elisa_example_data
##        a        d        c        b 
##  4.12406  0.04532 -5.31056  7.62972 
##  residual sum-of-squares: 0.02221
## 
## Number of iterations to convergence: 9 
## Achieved convergence tolerance: 1.49e-08</code></pre>
<pre><code>## 
## Formula: absorbance ~ ((a - d)/(1 + (log_dilution/c)^b)) + d
## 
## Parameters:
##   Estimate Std. Error  t value Pr(&gt;|t|)    
## a  4.12406    0.05820   70.860 1.75e-12 ***
## d  0.04532    0.02268    1.998   0.0808 .  
## c -5.31056    0.03933 -135.037 1.01e-14 ***
## b  7.62972    0.35854   21.280 2.50e-08 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## Residual standard error: 0.05269 on 8 degrees of freedom
## 
## Number of iterations to convergence: 9 
## Achieved convergence tolerance: 1.49e-08</code></pre>
</div>
<div id="apply-the-function-to-the-data" class="section level4" number="5.3.2.4">
<h4>
<span class="header-section-number">5.3.2.4</span> Apply the function to the data<a class="anchor" aria-label="anchor" href="#apply-the-function-to-the-data"><i class="fas fa-link"></i></a>
</h4>
</div>
<div id="plot-the-sigmoid-curve-with-fitted-sigmoid-model" class="section level4" number="5.3.2.5">
<h4>
<span class="header-section-number">5.3.2.5</span> Plot the sigmoid curve with fitted sigmoid model<a class="anchor" aria-label="anchor" href="#plot-the-sigmoid-curve-with-fitted-sigmoid-model"><i class="fas fa-link"></i></a>
</h4>
<div class="inline-figure"><img src="csu-impactb_files/figure-html/unnamed-chunk-44-1.png" width="672"></div>
</div>
</div>
</div>
<div id="endpoint-titer-method" class="section level2" number="5.4">
<h2>
<span class="header-section-number">5.4</span> 2. Endpoint titer method<a class="anchor" aria-label="anchor" href="#endpoint-titer-method"><i class="fas fa-link"></i></a>
</h2>
<p>The endpoint titer approach chooses an absorbance value just above the background noise (or the lower asymptotic level). <strong>The highest dilution with an absorbance greater than this predetermined value is the endpoint titer.</strong> This method is based on the assumption that a sample with a higher protein concentration will require a higher dilution factor to achieve an absorbance just above the level of background noise.</p>
<div id="create-an-endpoint-titer-function-and-apply-it-to-the-output-of-the-fitted-sigmoid-model-values." class="section level3" number="5.4.1">
<h3>
<span class="header-section-number">5.4.1</span> Create an endpoint titer function and apply it to the output of the fitted sigmoid model values.<a class="anchor" aria-label="anchor" href="#create-an-endpoint-titer-function-and-apply-it-to-the-output-of-the-fitted-sigmoid-model-values."><i class="fas fa-link"></i></a>
</h3>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##  -8.113  -8.113  -8.113  -8.113  -8.113  -8.113</code></pre>
<pre><code>## [1] -8.113285</code></pre>
</div>
<div id="other-methods-to-analyze-elisa-data" class="section level3" number="5.4.2">
<h3>
<span class="header-section-number">5.4.2</span> Other methods to analyze ELISA data<a class="anchor" aria-label="anchor" href="#other-methods-to-analyze-elisa-data"><i class="fas fa-link"></i></a>
</h3>
<div id="absorption-summation" class="section level4" number="5.4.2.1">
<h4>
<span class="header-section-number">5.4.2.1</span> Absorption summation<a class="anchor" aria-label="anchor" href="#absorption-summation"><i class="fas fa-link"></i></a>
</h4>
</div>
<div id="area-under-the-curve" class="section level4" number="5.4.2.2">
<h4>
<span class="header-section-number">5.4.2.2</span> Area under the curve<a class="anchor" aria-label="anchor" href="#area-under-the-curve"><i class="fas fa-link"></i></a>
</h4>
<p>In this model of data analysis, we sum all the absorbance values from each sample to obtain one value. This value is termed as absorption summation (AS). Using the above data, the AS will be calculated as below:</p>
<pre><code>## [1] 12.24</code></pre>
</div>
</div>
</div>
<div id="apply-the-fitting-sigmoid-model-and-endpoint-titer-function-in-our-dataset" class="section level2" number="5.5">
<h2>
<span class="header-section-number">5.5</span> Apply the fitting sigmoid model and endpoint titer function in our dataset<a class="anchor" aria-label="anchor" href="#apply-the-fitting-sigmoid-model-and-endpoint-titer-function-in-our-dataset"><i class="fas fa-link"></i></a>
</h2>
<p>The presented data is from a mouse study. In this data, presence of IgG antibody has been evaluated against receptor binding domain (RBD) of SARS-CoV-2 virus in two different groups of mice. We need to elucidate which group has higher concentration of the antibodies.</p>
<div id="read-in-the-data-1" class="section level4" number="5.5.0.1">
<h4>
<span class="header-section-number">5.5.0.1</span> Read in the data<a class="anchor" aria-label="anchor" href="#read-in-the-data-1"><i class="fas fa-link"></i></a>
</h4>
</div>
<div id="tidy-the-data" class="section level4" number="5.5.0.2">
<h4>
<span class="header-section-number">5.5.0.2</span> Tidy the data<a class="anchor" aria-label="anchor" href="#tidy-the-data"><i class="fas fa-link"></i></a>
</h4>
<pre><code>## # A tibble: 6 × 4
##   Groups  Dilution mouse_id absorbance
##   &lt;chr&gt;   &lt;chr&gt;    &lt;chr&gt;         &lt;dbl&gt;
## 1 Group 1 1/50     Mouse_1         4.1
## 2 Group 1 1/50     Mouse_2         3.9
## 3 Group 1 1/50     Mouse_3         4.3
## 4 Group 1 1/50     Mouse_4         4.2
## 5 Group 1 1/50     Mouse_5         4  
## 6 Group 1 1/100    Mouse_1         3.9</code></pre>
<pre><code>##    Groups numerator denomenator mouse_id absorbance dilution log_dilution
## 1 Group 1         1          50  Mouse_1        4.1     0.02    -5.643856
## 2 Group 1         1          50  Mouse_2        3.9     0.02    -5.643856
## 3 Group 1         1          50  Mouse_3        4.3     0.02    -5.643856
## 4 Group 1         1          50  Mouse_4        4.2     0.02    -5.643856
## 5 Group 1         1          50  Mouse_5        4.0     0.02    -5.643856
## 6 Group 1         1         100  Mouse_1        3.9     0.01    -6.643856</code></pre>
<div id="converting-data-into-dataframe" class="section level5" number="5.5.0.2.1">
<h5>
<span class="header-section-number">5.5.0.2.1</span> converting data into dataframe<a class="anchor" aria-label="anchor" href="#converting-data-into-dataframe"><i class="fas fa-link"></i></a>
</h5>
<pre><code>## # A tibble: 6 × 3
## # Groups:   Groups, mouse_id [6]
##   Groups  mouse_id data             
##   &lt;chr&gt;   &lt;chr&gt;    &lt;list&gt;           
## 1 Group 1 Mouse_1  &lt;tibble [10 × 5]&gt;
## 2 Group 1 Mouse_2  &lt;tibble [10 × 5]&gt;
## 3 Group 1 Mouse_3  &lt;tibble [10 × 5]&gt;
## 4 Group 1 Mouse_4  &lt;tibble [10 × 5]&gt;
## 5 Group 1 Mouse_5  &lt;tibble [10 × 5]&gt;
## 6 Group 2 Mouse_1  &lt;tibble [10 × 5]&gt;</code></pre>
</div>
<div id="plot-the-curves-to-evaluate-the-a-d-c-and-b" class="section level5" number="5.5.0.2.2">
<h5>
<span class="header-section-number">5.5.0.2.2</span> plot the curves to evaluate the a, d, c, and b<a class="anchor" aria-label="anchor" href="#plot-the-curves-to-evaluate-the-a-d-c-and-b"><i class="fas fa-link"></i></a>
</h5>
<div class="inline-figure"><img src="csu-impactb_files/figure-html/unnamed-chunk-50-1.png" width="768" style="display: block; margin: auto;"></div>
<p>Based on the curve, the values are:</p>
<p>a = 4,
d = 0
c = 2
b = 1</p>
</div>
</div>
<div id="creating-a-function-for-fitting-model" class="section level3" number="5.5.1">
<h3>
<span class="header-section-number">5.5.1</span> Creating a function for fitting model<a class="anchor" aria-label="anchor" href="#creating-a-function-for-fitting-model"><i class="fas fa-link"></i></a>
</h3>
<div id="fitting-the-model-into-the-dataset" class="section level4" number="5.5.1.1">
<h4>
<span class="header-section-number">5.5.1.1</span> Fitting the model into the dataset<a class="anchor" aria-label="anchor" href="#fitting-the-model-into-the-dataset"><i class="fas fa-link"></i></a>
</h4>
<pre><code>## Nonlinear regression model
##   model: absorbance ~ ((a - d)/(1 + (log_dilution/c)^b)) + d
##    data: df_elisa
##        a        d        c        b 
##   4.3070  -0.6009 -10.2577   5.2893 
##  residual sum-of-squares: 0.1199
## 
## Number of iterations to convergence: 7 
## Achieved convergence tolerance: 1.49e-08</code></pre>
</div>
<div id="apply-the-fitted-model-function-to-the-whole-dataframe" class="section level4" number="5.5.1.2">
<h4>
<span class="header-section-number">5.5.1.2</span> Apply the fitted model function to the whole dataframe<a class="anchor" aria-label="anchor" href="#apply-the-fitted-model-function-to-the-whole-dataframe"><i class="fas fa-link"></i></a>
</h4>
<pre><code>## # A tibble: 6 × 4
## # Groups:   Groups, mouse_id [6]
##   Groups  mouse_id data              fitted_data
##   &lt;chr&gt;   &lt;chr&gt;    &lt;list&gt;            &lt;list&gt;     
## 1 Group 1 Mouse_1  &lt;tibble [10 × 5]&gt; &lt;nls&gt;      
## 2 Group 1 Mouse_2  &lt;tibble [10 × 5]&gt; &lt;nls&gt;      
## 3 Group 1 Mouse_3  &lt;tibble [10 × 5]&gt; &lt;nls&gt;      
## 4 Group 1 Mouse_4  &lt;tibble [10 × 5]&gt; &lt;nls&gt;      
## 5 Group 1 Mouse_5  &lt;tibble [10 × 5]&gt; &lt;nls&gt;      
## 6 Group 2 Mouse_1  &lt;tibble [10 × 5]&gt; &lt;nls&gt;</code></pre>
</div>
<div id="take-out-the-summary-of-the-data" class="section level4" number="5.5.1.3">
<h4>
<span class="header-section-number">5.5.1.3</span> Take out the summary of the data<a class="anchor" aria-label="anchor" href="#take-out-the-summary-of-the-data"><i class="fas fa-link"></i></a>
</h4>
<pre><code>## Nonlinear regression model
##   model: absorbance ~ ((a - d)/(1 + (log_dilution/c)^b)) + d
##    data: df_elisa
##        a        d        c        b 
##   4.3070  -0.6009 -10.2577   5.2893 
##  residual sum-of-squares: 0.1199
## 
## Number of iterations to convergence: 7 
## Achieved convergence tolerance: 1.49e-08</code></pre>
</div>
</div>
</div>
<div id="create-function-of-fitted-model-and-endpoint-titer-where-the-output-of-the-fitted-model-data-will-be-the-input-of-the-endpoint-titer" class="section level2" number="5.6">
<h2>
<span class="header-section-number">5.6</span> Create function of Fitted model and endpoint titer, where the output of the fitted model data will be the input of the endpoint titer<a class="anchor" aria-label="anchor" href="#create-function-of-fitted-model-and-endpoint-titer-where-the-output-of-the-fitted-model-data-will-be-the-input-of-the-endpoint-titer"><i class="fas fa-link"></i></a>
</h2>
<div id="apply-the-fitted-model-fuction-into-the-nested-data-and-use-the-output-of-the-fitted-data-as-the-input-for-endpoint-titer-value-evaluation" class="section level4" number="5.6.0.1">
<h4>
<span class="header-section-number">5.6.0.1</span> Apply the fitted model fuction into the nested data and use the output of the fitted data as the input for endpoint titer value evaluation<a class="anchor" aria-label="anchor" href="#apply-the-fitted-model-fuction-into-the-nested-data-and-use-the-output-of-the-fitted-data-as-the-input-for-endpoint-titer-value-evaluation"><i class="fas fa-link"></i></a>
</h4>
<div id="run-fitted-model-on-the-data" class="section level5" number="5.6.0.1.1">
<h5>
<span class="header-section-number">5.6.0.1.1</span> Run fitted model on the data<a class="anchor" aria-label="anchor" href="#run-fitted-model-on-the-data"><i class="fas fa-link"></i></a>
</h5>
<pre><code>## # A tibble: 6 × 4
## # Groups:   Groups, mouse_id [6]
##   Groups  mouse_id data              fitted_data
##   &lt;chr&gt;   &lt;chr&gt;    &lt;list&gt;            &lt;list&gt;     
## 1 Group 1 Mouse_1  &lt;tibble [10 × 5]&gt; &lt;nls&gt;      
## 2 Group 1 Mouse_2  &lt;tibble [10 × 5]&gt; &lt;nls&gt;      
## 3 Group 1 Mouse_3  &lt;tibble [10 × 5]&gt; &lt;nls&gt;      
## 4 Group 1 Mouse_4  &lt;tibble [10 × 5]&gt; &lt;nls&gt;      
## 5 Group 1 Mouse_5  &lt;tibble [10 × 5]&gt; &lt;nls&gt;      
## 6 Group 2 Mouse_1  &lt;tibble [10 × 5]&gt; &lt;nls&gt;</code></pre>
</div>
<div id="taking-output-of-the-fitted-model-function-and-into-endpoint-titer-function" class="section level5" number="5.6.0.1.2">
<h5>
<span class="header-section-number">5.6.0.1.2</span> Taking output of the fitted model function and into endpoint titer function<a class="anchor" aria-label="anchor" href="#taking-output-of-the-fitted-model-function-and-into-endpoint-titer-function"><i class="fas fa-link"></i></a>
</h5>
</div>
</div>
<div id="plot-the-endpoint-titer-data-for-the-two-groups" class="section level4" number="5.6.0.2">
<h4>
<span class="header-section-number">5.6.0.2</span> Plot the endpoint titer data for the two groups<a class="anchor" aria-label="anchor" href="#plot-the-endpoint-titer-data-for-the-two-groups"><i class="fas fa-link"></i></a>
</h4>
<div class="inline-figure"><img src="csu-impactb_files/figure-html/unnamed-chunk-58-1.png" width="672"></div>
</div>
<div id="perform-statistical-analysis-on-the-data" class="section level4" number="5.6.0.3">
<h4>
<span class="header-section-number">5.6.0.3</span> Perform statistical analysis on the data<a class="anchor" aria-label="anchor" href="#perform-statistical-analysis-on-the-data"><i class="fas fa-link"></i></a>
</h4>
<pre><code>## # A tibble: 1 × 10
##   estimate estimate1 estimate2 statistic    p.value parameter conf.low conf.high
##      &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;      &lt;dbl&gt;     &lt;dbl&gt;    &lt;dbl&gt;     &lt;dbl&gt;
## 1   -0.800     -14.0     -13.2     -18.8 0.00000268      5.63   -0.906    -0.695
## # ℹ 2 more variables: method &lt;chr&gt;, alternative &lt;chr&gt;</code></pre>
</div>
<div id="statistical-data-analysis-for-more-than-two-groups" class="section level4" number="5.6.0.4">
<h4>
<span class="header-section-number">5.6.0.4</span> Statistical data analysis for more than two groups<a class="anchor" aria-label="anchor" href="#statistical-data-analysis-for-more-than-two-groups"><i class="fas fa-link"></i></a>
</h4>
</div>
</div>
<div id="elisa-data-processing" class="section level2" number="5.7">
<h2>
<span class="header-section-number">5.7</span> ELISA data processing<a class="anchor" aria-label="anchor" href="#elisa-data-processing"><i class="fas fa-link"></i></a>
</h2>
<p>We read ELISA plate in a 96 well plate using a plate reader. The plate reader generates the data in form of number in an excel sheet. We have created this pipeline/worksheet to bring out the information from the excl sheet to a tidy format in which the above created fitted model and endpoint titer functions can be applied.</p>
<div id="read-in-the-first-dataset" class="section level4" number="5.7.0.1">
<h4>
<span class="header-section-number">5.7.0.1</span> Read in the first dataset<a class="anchor" aria-label="anchor" href="#read-in-the-first-dataset"><i class="fas fa-link"></i></a>
</h4>
<p>Below is the example ELISA data that has came straight out of the plate reader. This data is arranged in a 96-well plate format and contains Optical Density (OD) values.</p>
<pre><code>## # A tibble: 6 × 12
##   ...1          ...2  ...3 ...4   ...5  ...6 ...7   ...8  ...9 ...10 ...11 ...12
##   &lt;chr&gt;        &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt; &lt;dbl&gt; &lt;dbl&gt;
## 1 5.199999999… 0.05  0.069 6.3E… 0.061 0.122 0.16… 0.145 0.135 6.80… 0.053 0.05 
## 2 7.900000000… 0.098 0.069 6.80… 0.115 0.202 5.89… 0.134 0.069 0.106 0.05  0.075
## 3 8.899999999… 0.133 0.119 OVRF… 3.87  2.32  OVRF… 3.85  2.12  OVRF… 3.21  1.02 
## 4 OVRFLW       3.46  1.16  OVRF… 3.80  2.36  OVRF… 3.70  1.49  OVRF… 3.68  1.63 
## 5 3.815999999… 1.82  0.446 3.89… 3.42  1.13  OVRF… 2.33  0.608 OVRF… 3.41  1.10 
## 6 OVRFLW       3.69  1.43  OVRF… 3.66  1.27  3.839 1.74  0.444 2.49… 0.637 0.704</code></pre>
</div>
<div id="tidy-dataset-1" class="section level4" number="5.7.0.2">
<h4>
<span class="header-section-number">5.7.0.2</span> Tidy dataset 1<a class="anchor" aria-label="anchor" href="#tidy-dataset-1"><i class="fas fa-link"></i></a>
</h4>
<p>It is important to clean the data and arrange it in a format on which we can apply formulas and functions.</p>
<pre><code>## # A tibble: 6 × 2
##   well_id od_450nm
##   &lt;chr&gt;      &lt;dbl&gt;
## 1 A1         0.052
## 2 A2         0.05 
## 3 A3         0.069
## 4 A4         0.063
## 5 A5         0.061
## 6 A6         0.122</code></pre>
</div>
<div id="read-in-the-second-data-set" class="section level4" number="5.7.0.3">
<h4>
<span class="header-section-number">5.7.0.3</span> Read in the second data set<a class="anchor" aria-label="anchor" href="#read-in-the-second-data-set"><i class="fas fa-link"></i></a>
</h4>
<p>The second dataset contains the information such as groups, mouse id, and dilutions for the respective wells of the 96 well plate for the dataset-1.</p>
<pre><code>## # A tibble: 6 × 12
##   ...1        ...2   ...3  ...4  ...5  ...6  ...7  ...8  ...9  ...10 ...11 ...12
##   &lt;chr&gt;       &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;
## 1 blank       secon… naïv… 1A-1… 1A-1… 1A-1… 1A-2… 1A-2… 1A-2… 1A-3… 1A-3… 1A-3…
## 2 1A-4 (1/250 1A-4 … 1A-4… 1B-1… 1B-1… 1B-1… 1B-2… 1B-2… 1B-2… 1B-3… 1B-3… 1B-3…
## 3 1B-4 (1/250 1B-4 … 1B-4… 2A-1… 2A-1… 2A-1… 2A-2… 2A-2… 2A-2… 2A-3… 2A-3… 2A-3…
## 4 2B-1 (1/250 2B-1 … 2B-1… 2B-2… 2B-2… 2B-2… 2B-3… 2B-3… 2B-3… 2B-4… 2B-4… 2B-4…
## 5 3A-1 (1/250 3A-1 … 3A-1… 3A-2… 3A-2… 3A-2… 3A-3… 3A-3… 3A-3… 3A-4… 3A-4… 3A-4…
## 6 3B-1 (1/250 3B-1 … 3B-1… 3B-2… 3B-2… 3B-2… 3B-3… 3B-3… 3B-3… 3B-4… 3B-4… 3B-4…</code></pre>
</div>
<div id="tidy-dataset-2" class="section level4" number="5.7.0.4">
<h4>
<span class="header-section-number">5.7.0.4</span> Tidy dataset-2<a class="anchor" aria-label="anchor" href="#tidy-dataset-2"><i class="fas fa-link"></i></a>
</h4>
<pre><code>## # A tibble: 6 × 2
##   well_id information  
##   &lt;chr&gt;   &lt;chr&gt;        
## 1 A1      blank        
## 2 A2      secondary    
## 3 A3      naïve (1/250)
## 4 A4      1A-1 (1/250  
## 5 A5      1A-1 (1/1250 
## 6 A6      1A-1 (1/6250</code></pre>
</div>
<div id="merge-dataset-1-with-od-information-with-dataset-2-with-respective-data-information" class="section level4" number="5.7.0.5">
<h4>
<span class="header-section-number">5.7.0.5</span> Merge dataset-1 (with OD information) with dataset-2 (with respective data information)<a class="anchor" aria-label="anchor" href="#merge-dataset-1-with-od-information-with-dataset-2-with-respective-data-information"><i class="fas fa-link"></i></a>
</h4>
<p>To create a complete full dataset with Groups, mouse-id, dilutions, and OD, we merged the dataset-1 and dataset-2 together. We also cleaned the data set so that mouse-ID and dilution columns are separate and have their own columns.</p>
<pre><code>## # A tibble: 6 × 3
##   well_id od_450nm information  
##   &lt;chr&gt;      &lt;dbl&gt; &lt;chr&gt;        
## 1 A1         0.052 blank        
## 2 A2         0.05  secondary    
## 3 A3         0.069 naïve (1/250)
## 4 A4         0.063 1A-1 (1/250  
## 5 A5         0.061 1A-1 (1/1250 
## 6 A6         0.122 1A-1 (1/6250</code></pre>
<pre><code>## # A tibble: 6 × 4
##   well_id od_450nm sample_id   dilution
##   &lt;chr&gt;      &lt;dbl&gt; &lt;chr&gt;       &lt;chr&gt;   
## 1 A1         0.052 "blank"     &lt;NA&gt;    
## 2 A2         0.05  "secondary" &lt;NA&gt;    
## 3 A3         0.069 "naïve "    1/250)  
## 4 A4         0.063 "1A-1 "     1/250   
## 5 A5         0.061 "1A-1 "     1/1250  
## 6 A6         0.122 "1A-1 "     1/6250</code></pre>
<pre><code>## # A tibble: 6 × 4
##   well_id sample_id   dilution od_450nm
##   &lt;chr&gt;   &lt;chr&gt;          &lt;dbl&gt;    &lt;dbl&gt;
## 1 A1      "blank"           NA    0.052
## 2 A2      "secondary"       NA    0.05 
## 3 A3      "naïve "         250    0.069
## 4 A4      "1A-1 "          250    0.063
## 5 A5      "1A-1 "         1250    0.061
## 6 A6      "1A-1 "         6250    0.122</code></pre>

</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="colony-forming-units-to-determine-bacterial-counts.html"><span class="header-section-number">4</span> Colony forming units to determine bacterial counts</a></div>
<div class="next"><a href="flow-cytometry.html"><span class="header-section-number">6</span> Flow cytometry</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#enzyme-linked-immunosorbest-assay-elisa"><span class="header-section-number">5</span> Enzyme-linked immunosorbest assay (ELISA)</a></li>
<li>
<a class="nav-link" href="#importance-of-elisa"><span class="header-section-number">5.1</span> Importance of ELISA</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#principle-of-elisa"><span class="header-section-number">5.1.1</span> Principle of ELISA</a></li>
<li><a class="nav-link" href="#loading-libraries"><span class="header-section-number">5.1.2</span> Loading libraries</a></li>
</ul>
</li>
<li><a class="nav-link" href="#elisa-data-analysis"><span class="header-section-number">5.2</span> ELISA data analysis</a></li>
<li>
<a class="nav-link" href="#curve-fitting-model"><span class="header-section-number">5.3</span> 1. Curve fitting model:</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#how-do-we-perform-curve-fitting-model"><span class="header-section-number">5.3.1</span> How do we perform curve fitting model</a></li>
<li><a class="nav-link" href="#an-example-of-the-curve-fitting-model"><span class="header-section-number">5.3.2</span> An example of the curve fitting model</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#endpoint-titer-method"><span class="header-section-number">5.4</span> 2. Endpoint titer method</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#create-an-endpoint-titer-function-and-apply-it-to-the-output-of-the-fitted-sigmoid-model-values."><span class="header-section-number">5.4.1</span> Create an endpoint titer function and apply it to the output of the fitted sigmoid model values.</a></li>
<li><a class="nav-link" href="#other-methods-to-analyze-elisa-data"><span class="header-section-number">5.4.2</span> Other methods to analyze ELISA data</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#apply-the-fitting-sigmoid-model-and-endpoint-titer-function-in-our-dataset"><span class="header-section-number">5.5</span> Apply the fitting sigmoid model and endpoint titer function in our dataset</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#creating-a-function-for-fitting-model"><span class="header-section-number">5.5.1</span> Creating a function for fitting model</a></li></ul>
</li>
<li><a class="nav-link" href="#create-function-of-fitted-model-and-endpoint-titer-where-the-output-of-the-fitted-model-data-will-be-the-input-of-the-endpoint-titer"><span class="header-section-number">5.6</span> Create function of Fitted model and endpoint titer, where the output of the fitted model data will be the input of the endpoint titer</a></li>
<li><a class="nav-link" href="#elisa-data-processing"><span class="header-section-number">5.7</span> ELISA data processing</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>ImpactTB/BAA: Standard Operating Procedures for Data Analysis</strong>" was written by Colorado State University Coding Team. It was last built on 2024-06-21.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
